src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Type/TypeTests.cs

```diff
@@ -513,6 +513,19 @@ public void GetTypeByName_ValidType_ReturnsExpected(string typeName, Type expect
             Assert.Equal(expectedType, Type.GetType(typeName.ToLower(), throwOnError: false, ignoreCase: true));
         }
 
+        public static IEnumerable<object[]> GetTypeByName_Invalid_ClrArguments()
+        {
+            Type expectedException = PlatformDetection.IsMonoRuntime
+                ? typeof(ArgumentException) // https://github.com/dotnet/runtime/issues/45033
+                : typeof(TypeLoadException);
+
+            yield return new object[] { "System.Int32&&", expectedException, true };
+            yield return new object[] { "System.Int32&*", expectedException, true };
+            yield return new object[] { "System.Int32&[]", expectedException, true };
+            yield return new object[] { "System.Int32&[*]", expectedException, true };
+            yield return new object[] { "System.Int32&[,]", expectedException, true };
```

Comment:
```suggestion
            yield return new object[] { "System.Int32&[,]", expectedException, true };
            yield return new object[] { "System.Void[]", expectedException, true };
            yield return new object[] { "System.TypedReference[]", expectedException, true };
```
More interesting cases